package.json

```diff
@@ -38,14 +38,19 @@
     "test:p2p:watch": "mocha --timeout 10000 -r ts-node/register test/p2p/*  --watch --watch-extensions ts",
     "test:crypto": "mocha --timeout 10000 -r ts-node/register test/crypto/*",
     "test:sim": "(npm run test:sim:build && npm run test:sim:run)",
-    "test:sim:build": "(npm run test:sim:update-xud && cd test/simulation && ./docker-build.sh)",
+    "test:sim:build": "(npm run test:sim:compile:xud && npm run test:sim:compile:custom-xud && cd test/simulation && ./docker-build.sh)",
     "test:sim:build:test": "(cd test/simulation && ./docker-build.sh test)",
-    "test:sim:build:xud": "(npm run test:sim:update-xud && cd test/simulation && ./docker-build.sh xud)",
+    "test:sim:build:xud": "(npm run test:sim:compile:xud && npm run test:sim:compile:custom-xud && cd test/simulation && ./docker-build.sh xud)",
     "test:sim:build:lnd": "(cd test/simulation && ./docker-build.sh lnd)",
-    "test:sim:run": "(cd test/simulation && ./docker-run.sh TestIntegration)",
+    "test:sim:run": "(cd test/simulation && ./docker-run.sh)",
+    "test:sim:run:integration": "(cd test/simulation && ./docker-run.sh TestIntegration)",
     "test:sim:run:instability": "(cd test/simulation && ./docker-run.sh TestInstability)",
     "test:sim:run:security": "(cd test/simulation && ./docker-run.sh TestSecurity)",
-    "test:sim:update-xud": "(npm run compile:dev && mkdir -p test/simulation/docker-xud/xud && cp -a dist test/simulation/docker-xud/xud && cp -a bin test/simulation/docker-xud/xud && cp package.json test/simulation/docker-xud/xud)",
+    "test:sim:compile:xud": "(npm run compile && mkdir -p test/simulation/docker-xud/xud && cp -a dist test/simulation/docker-xud/xud && cp -a bin test/simulation/docker-xud/xud && cp package.json test/simulation/docker-xud/xud)",
```

Comment:
Will there ever be a reason to run `compile:xud` and `compile:custom-xud` on their own? It seems to me like it's always going to in connection with `build:xud`, so I'm wondering if it makes sense to get rid of these scripts and just incorporate these commands direclty into `build:xud`. It would make the `build:xud` script quite long, but I think as it's a little confusing to have both `test:sim:compile:xud` and `test:sim:build:xud` since those are nearly identical in naming, and the README instructions don't reference any of the `test:sim:compile` scripts so they're currently undocumented.

Reply:
No, there's no real reason to run them on their own. But they are part of both `test:sim:build` and `test:sim:build:xud`, so writing them inline will make things unreadable/maintainable IMO (specially because `test:sim:compile:custom-xud` is quite long). 
I would have moved them to shell scripts under `test/simulation`, but they have dependency on other npm scripts.
I would have added them to README, but they are not supposed to be used directly.

So in the overall, I leaving them as-is is the least bad option for now.
Best solution IMO would be to document the npm scripts.